### PR TITLE
feat: builder docker/node and sdk-js

### PR DESCRIPTION
**Pull request by hacdias**
This PR is aligned with the efforts in https://github.com/testground/sdk-js/issues/1.

- Adds a `docker/node` builder with the following base Dockerfile:
	```docker
	ARG BASE_IMAGE
	FROM ${BASE_IMAGE} AS builder
	ENV PLAN_DIR /plan
	WORKDIR /plan
	COPY . /
	RUN npm ci
	EXPOSE 6060
	ENTRYPOINT [ "npm", "start"]
	```
- Adds a test plan `example-js` with the following tests:
	- pingpong
	- sync
	- failure
	- output (not complete because metrics aren't finished yet)
- Adds `example-js/pingpong` to CI

### CODE REVIEW

**Comment by hacdias** [comment]
We have to make [`network.Config`](https://github.com/testground/sdk-go/pull/34/files) the most compatible possible. Next week, I will tackle on what @raulk mentioned in the linked PR and will also implement an Unmarshaller/Decoder for that struct.

Some things such as `net.IPNet` are not easily converted to the same thing as Go. My suggestion is to just pass around the string CIDR. That will virtually work on all implementations since there's always a function to convert CIDRs to native structs.

We can keep using `net.IPNet` in Go but I'll need to implement an unmarshaller.

**Comment by raulk** [comment]
Yeah, I think struct tags will do for the casing. And for IPNet, there's actually a marshaller/unmarshaller you can use. Let me try digging it up!

**Comment by raulk** [comment]
Here it is: https://github.com/testground/sdk-go/blob/master/runtime/runparams.go

**Comment by hacdias** [comment]
Just throwing it out again: I think we should collect all this types in some place! 😆

**Comment by hacdias** [comment]
I'm changing the base of this PR to #1154 because that one includes the newer SDK with the updates to all tests. However, I'd prefer if we could merge #1154 into master when possible, since that one is not even based on master.

**Comment by hacdias** [comment]
Update: ping-pong test working!

**Comment by hacdias** [comment]
@nonsense @raulk I feel this PR is finished. Even though sdk-js isn't complete (missing metrics before releasing 1.0), I feel this PR is ready for review and should be merged. Later, we can make further changes if needed.

File: pkg/build/docker_node.go [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+	}
+
+	cliopts := []client.Opt{client.FromEnv, client.WithAPIVersionNegotiation()}
+
+	var (
+		basesrc  = in.UnpackedSources.BaseDir
+		cli, err = client.NewClientWithOpts(cliopts...)
+	)
+
+	ctx, cancel := context.WithTimeout(ctx, 30*time.Minute)
```
Comment (nonsense) [inline pkg/build/docker_node.go:45]: I see this is copied from `docker:go` builder, but we don't need it anymore, as we are always passing through the supervisor, which applies the `task timeout` (by default 10min, but configurable). So I suggest we delete this (maybe also directly delete in `docker:go` builder as well.

File: pkg/build/docker_node.go [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+	defer cancel()
```
Comment (nonsense) [inline pkg/build/docker_node.go:46]:
```suggestion
```

File: pkg/build/docker_node.go [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+
+	if err != nil {
+		return nil, err
+	}
```
Comment (nonsense) [inline pkg/build/docker_node.go:50]: Nitpick, but I find it weird that this error check is separated from the declaration of `cli`... not sure why we want to group `basesrc` and `cli` under the same `var` group.

File: plans/example-js/pingpong.js [hunk 4]
```diff
@@ -0,0 +1,186 @@
...
+    const oldAddrs = os.networkInterfaces()
+
+    const config = {
+      network: 'default',
+      enable: true,
+      default: {
+        latency: 100 * 1000 * 1000, // 100ms in nanoseconds
+        bandwidth: 1 << 20 // 1 Mib
+      },
+      callbackState: 'network-configured', // Fix this small and big letters thingies.
```
Comment (nonsense) [inline plans/example-js/pingpong.js:32]: You have fixed that, right? Update comment?

File: plans/example-js/pingpong.js [hunk 5]
```diff
@@ -0,0 +1,186 @@
...
+
+    // Make sure that the IP addresses don't change unless we request it.
+    const newAddrs = os.networkInterfaces()
+    if (!sameAddrs(oldAddrs, newAddrs)) {
+      throw new Error('interfaces changed')
+    }
+
+    runenv.recordMessage(`I am ${seq}`)
+
+    const ipC = (seq >> 8) + 1
```
Comment (nonsense) [inline plans/example-js/pingpong.js:49]: Considering that `ping-pong` is always using 2 instances (and doesn't work with more than 2 instances), this is a really weird way to update the C and D parts of the IP. I know this is copying the network/ping-pong Go impl., just saying we might want to refactor this soon :)

**Review by nonsense — APPROVED** [review]
Overall LGTM!

**Comment by hacdias** [comment]
I'm going to address the comments and then merge!

File: pkg/build/docker_node.go (hunk 3 above)
Comment (hacdias) [reply pkg/build/docker_node.go:44]: Separated! I didn't think much of that. I used `docker:go` as a boilerplate and basically just removed what wasn't needed. MIssed that!

**Pull request merged**